optimization.py

```diff
@@ -163,7 +273,25 @@ def apply_gradients(self, grads_and_vars, global_step=None, name=None):
       if self._do_use_weight_decay(param_name):
         update += self.weight_decay_rate * param
 
-      update_with_lr = self.learning_rate * update
+      ############## BELOW ARE THE SPECIFIC PARTS FOR LAMB ##############
+
+      # Note: Here are two choices for scaling function \phi(z)
+      # minmax:   \phi(z) = min(max(z, \gamma_l), \gamma_u)
+      # identity: \phi(z) = z
+      # The authors does not mention what is \gamma_l and \gamma_u
+      # UPDATE: after asking authors, they provide me the code below.
+      # ratio = array_ops.where(math_ops.greater(w_norm, 0), array_ops.where(
+      #      math_ops.greater(g_norm, 0), (w_norm / g_norm), 1.0), 1.0)
+
```

Comment:
I don't think the comments about the authors and the update is necessary as long the information is there but upto you.